client/matchmaking/matchmaking-reducer.ts

```diff
@@ -1,52 +1,40 @@
 import { List, Map, Record, Set } from 'immutable'
-import {
-  MAPS_TOGGLE_FAVORITE,
-  MATCHMAKING_ACCEPT,
-  MATCHMAKING_CANCEL,
-  MATCHMAKING_FIND,
-  MATCHMAKING_GET_CURRENT_MAP_POOL,
-  MATCHMAKING_GET_CURRENT_MAP_POOL_BEGIN,
-  MATCHMAKING_UPDATE_ACCEPT_MATCH_FAILED,
-  MATCHMAKING_UPDATE_ACCEPT_MATCH_TIME,
-  MATCHMAKING_UPDATE_COUNTDOWN_START,
-  MATCHMAKING_UPDATE_COUNTDOWN_TICK,
-  MATCHMAKING_UPDATE_GAME_STARTED,
-  MATCHMAKING_UPDATE_GAME_STARTING,
-  MATCHMAKING_UPDATE_LOADING_CANCELED,
-  MATCHMAKING_UPDATE_MATCH_ACCEPTED,
-  MATCHMAKING_UPDATE_MATCH_FOUND,
-  MATCHMAKING_UPDATE_MATCH_READY,
-  NETWORK_SITE_CONNECTED,
-} from '../actions'
+import { MapInfoJson } from '../../common/maps'
+import { MatchmakingType } from '../../common/matchmaking'
+import { MAPS_TOGGLE_FAVORITE, NETWORK_SITE_CONNECTED } from '../actions'
 import { MapRecord } from '../maps/maps-reducer'
+import { FetchError } from '../network/fetch-action-types'
 import { keyedReducer } from '../reducers/keyed-reducer'
 
-export const Player = new Record({
-  id: null,
-  name: null,
-  race: null,
+export class MatchmakingPlayerRecord extends Record({
+  id: 0,
+  name: '',
+  race: 'r',
   rating: -1,
-})
-const Match = new Record({
+}) {}
+
+export class MatchmakingMatchRecord extends Record({
   numPlayers: 0,
   acceptedPlayers: 0,
-  type: null,
-  players: new List(),
-  preferredMaps: new Set(),
-  randomMaps: new Set(),
-  chosenMap: null,
-})
-const MapPool = new Record({
-  id: null,
-  type: null,
-  startDate: null,
-  maps: new List(),
-  byId: new Map(),
+  type: MatchmakingType.Match1v1,
+  players: List<MatchmakingPlayerRecord>(),
+  preferredMaps: Set<MapInfoJson>(),
+  randomMaps: Set<MapInfoJson>(),
+  chosenMap: undefined as MapInfoJson | undefined,
+}) {}
+
+export class MapPoolRecord extends Record({
+  id: '',
+  type: MatchmakingType.Match1v1,
+  startDate: new Date(),
+  maps: List<string>(),
+  byId: Map<string, MapInfoJson>(),
```

Comment:
You're storing records in here, so that's not an accurate typing. At the very least this should be `Readonly<MapInfoJson>`, but even that is fairly fake/not true and you'd be better off using `MapRecord`, which is what you're actually storing

Reply:
I also don't really know why we're storing the map info here at all, we already have a reducer that stores the map info by ID (`maps2-reducer`), you should really be handling the appropriate actions in there and only storing IDs here

Reply:
Gonna just fix the types for now, and then I'll move to using map IDs similar to what we do with users in a different PR.